examples/applications/object_detection/retina_net/basic/pascal_voc/train.py

```diff
@@ -0,0 +1,341 @@
+    boundaries=learning_rate_boundaries, values=learning_rates
+)
+
+optimizer = optimizers.SGD(
+    learning_rate=learning_rate_fn, momentum=0.9, global_clipnorm=10.0
+)
+
+"""
+## COCO metrics monitoring
+
+KerasCV offers a suite of in-graph COCO metrics that support batch-wise evaluation.
+More information on these metrics is available in:
+
+- [Efficient Graph-Friendly COCO Metric Computation for Train-Time Model Evaluation](https://arxiv.org/abs/2207.12120)
+- [Using KerasCV COCO Metrics](https://keras.io/guides/keras_cv/coco_metrics/)
+
+Lets construct two COCO metrics, an instance of
+`keras_cv.metrics.COCOMeanAveragePrecision` with the parameterization to match the
+standard COCO Mean Average Precision metric, and `keras_cv.metrics.COCORecall`
+parameterized to match the standard COCO Recall metric.
+"""
+
+metrics = [
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        class_ids=range(20),
+        bounding_box_format="xywh",
+        name="Mean Average Precision",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(20),
+        bounding_box_format="xywh",
+        max_detections=100,
+        name="Recall",
+    ),
+]
+
+"""
+## Training our model
+
+All that is left to do is train our model.  KerasCV object detection models follow the
+standard Keras workflow, leveraging `compile()` and `fit()`.
+
+Let's compile our model:
+"""
+
+loss = keras_cv.losses.ObjectDetectionLoss(
+    classes=20,
+    classification_loss=keras_cv.losses.FocalLoss(from_logits=True, reduction="none"),
+    box_loss=keras_cv.losses.SmoothL1Loss(cutoff=1.0, reduction="none"),
+    reduction="auto",
+)
+
+model.compile(
+    loss=loss,
+    optimizer=optimizer,
+    metrics=metrics,
+)
+
+"""
+All that is left to do is construct some callbacks:
+"""
+
+callbacks = [
+    callbacks_lib.TensorBoard(log_dir="logs"),
+    callbacks_lib.EarlyStopping(patience=30),
+]
+if FLAGS.wandb:
+    callbacks += [
+        wandb.keras.WandbCallback(save_model=False),
+    ]
+
+"""
+And run `model.fit()`!
+"""
+
+model.fit(
+    train_ds,
+    validation_data=val_ds,
+    epochs=FLAGS.epochs,
+    callbacks=callbacks,
+)
+
+"""
+## Results and conclusions
+
+KerasCV makes it easy to construct state-of-the-art object detection pipelines.  All of
+the KerasCV object detection components can be used independently, but also have deep
+integration with each other.  With KerasCV, bounding box augmentation, train-time COCO
+metrics evaluation, and more, are all made simple and consistent.
+
+By default, this script runs for a single epoch.  To run training to convergence,
+invoke the script with a command line flag `--epochs=500`.  To save you the effort of
+running the script for 500 epochs, we have produced a Weights and Biases report covering
+the training results below!  As a bonus, the report includes a training run with and
+without data augmentation.
+
+TODO(lukewood): add link to the WandB run once preprocessing bugs are resolved.
```

Comment:
I think you have a WandB run ready to link here now?

Reply:
I'd rather wait until we have strong scores to publish before putting osmething bad out.